litestar/app.py

```diff
@@ -388,7 +388,13 @@ def __init__(
 
         self._openapi_schema: OpenAPI | None = None
         self._debug: bool = True
+        self.stores: StoreRegistry = (
+            config.stores if isinstance(config.stores, StoreRegistry) else StoreRegistry(config.stores)
+        )
         self._lifespan_managers = config.lifespan
+        for store in self.stores._stores.values():
+            if store.handle_client_shutdown:
+                self._lifespan_managers.append(store)
```

Comment:
I think my earlier suggestion was bad. 
What I meant was:

- Always add the stores as lifespan managers here
- Handle the `handle_client_shutdown` inside `RedisStore.__aexit__`

That way each store can decide how it wants to handle shutdown if at all

Reply:
gotcha, I'll take that back later on, you obviously know in and out the codebase, that's very impressive, the LLM-joke was meant as a compliment by the way !

Reply:
> I think my earlier suggestion was bad. What I meant was:
> 
> * Always add the stores as lifespan managers here
> * Handle the `handle_client_shutdown` inside `RedisStore.__aexit__`
> 
> That way each store can decide how it wants to handle shutdown if at all

done, it's better since it clears those 46 breaks in tests by keeping existing stores the same.

also, the current test is like....complicated, but in the meantime the fact it popped up for me is exactly how we figured the closing wasn't performed, so I'd be tempted to keep it as is, but happ

time for some linter stuff now